src/parser/jobs/blm/modules/RotationWatchdog.tsx

```diff
@@ -0,0 +1,563 @@
+				</Trans>,
+				severity: this.thunder3Casts > 2 * maxThunders ? SEVERITY.MAJOR : SEVERITY.MEDIUM,
+				why: <Trans id="blm.rotation-watchdog.suggestions.excess-thunder.why">
+					At least <Plural value={this.thunder3Casts - maxThunders} one="# extra Thunder III was" other="# extra Thunder III were"/> cast.
+				</Trans>,
+			}))
+		}
+
+		// Checklist item for keeping Thunder 3 DoT rolling
+		this.checklist.add(new Rule({
+			name: <Trans id="blm.rotation-watchdog.checklist.dots.name">Keep your <StatusLink {...STATUSES.THUNDER_III} /> DoT up</Trans>,
+			description: <Trans id="blm.rotation-watchdog.checklist.dots.description">
+				Your <StatusLink {...STATUSES.THUNDER_III} /> DoT contributes significantly to your overall damage, both on its own, and from additional <StatusLink {...STATUSES.THUNDERCLOUD} /> procs. Try to keep the DoT applied.
+			</Trans>,
+			target: 95,
+			requirements: [
+				new Requirement({
+					name: <Fragment><StatusLink {...STATUSES.THUNDER_III} /> uptime</Fragment>,
+					percent: () => this.getThunderUptime(),
+				}),
+			],
+		}))
+	}
+
+	// Complete the previous cycle and start a new one
+	private startRecording(event: CastEvent) {
+		this.stopRecording(event)
+		this.currentRotation = new Cycle(event.timestamp, this.currentGaugeState)
+	}
+
+	// End the current cycle, send it off to error processing, and add it to the history list
+	private stopRecording(event: CastEvent | undefined) {
+		this.currentRotation.endTime = this.parser.currentTimestamp
+
+		// If an event object wasn't passed, or the event was a transpose that occurred during downtime,
+		// treat this as a rotation that ended with some kind of downtime
+		if (!event || (event && event.ability.guid === ACTIONS.TRANSPOSE.id &&
+			this.invuln.isUntargetable('all', event.timestamp))) {
+			this.currentRotation.finalOrDowntime = true
+		}
+
+		this.processCycle(this.currentRotation)
+		this.history.push(this.currentRotation)
+	}
+
+	// Process errors for this cycle
+	// TODO: Handle aoe things?
+	// TODO: Handle Flare?
+	private processCycle(currentRotation: Cycle) {
+		// Only process errors for rotations with more than the minimum number of casts,
+		// since fewer than that usually indicates something weird happening
+		if (currentRotation.casts.length <= MIN_ROTATION_LENGTH) {
+			currentRotation.errorCode = ERROR_CODES.SHORT
+			return
+		}
+
+		// Check for errors that apply for all cycles
+
+		// Check if the rotation included the expected number of Despair casts
+		if (currentRotation.missingDespairs) {
+			this.astralFiresMissingDespairs++
+			currentRotation.errorCode = ERROR_CODES.MISSING_DESPAIRS
+		}
+
+		// Check whether manafont was used before despair
+		if (currentRotation.hasManafont && currentRotation.actualDespairs > 0 &&
```

Comment:
indentation seems to have crapped out here completely. someone make me fix the linter already

Reply:
Yes it did @.@ Fix'd